controller/scheduler/k8s.py

```diff
@@ -0,0 +1,636 @@
+        try:
+            self._scale_app(name, num)
+        except Exception as e:
+            self._scale_app(name, old_replicas)
+            err = '{} (Scale): {}'.format(name, e)
+            raise RuntimeError(err)
+
+    def _create_rc(self, name, image, command, **kwargs):
+        container_fullname = name
+        app_name = kwargs.get('aname', {})
+        app_type = name.split(".")[1]
+        container_name = app_name+"-"+app_type
+        name = name.split(".")[0]
+        name = name.replace("_", "-")
+        args = command.split()
+
+        num = kwargs.get('num', {})
+        l = {}
+        l["name"] = name
+        l["id"] = app_name
+        l["appversion"] = kwargs.get('version', {})
+        l["version"] = self.apiversion
+        l["image"] = self.registry+"/"+image
+        l['num'] = num
+        l['containername'] = container_name
+        template = string.Template(RC_TEMPLATE).substitute(l)
+        js_template = json.loads(template)
+        containers = js_template["spec"]["template"]["spec"]["containers"]
+        containers[0]['args'] = args
+        loc = locals().copy()
+        loc.update(re.match(MATCH, container_fullname).groupdict())
+        mem = kwargs.get('memory', {}).get(loc['c_type'])
+        cpu = kwargs.get('cpu', {}).get(loc['c_type'])
+        if mem or cpu:
+            containers[0]["resources"] = {"limits": {}}
+        if mem:
+            if mem[-2:-1].isalpha() and mem[-1].isalpha():
+                mem = mem[:-1]
+            mem = mem+"i"
+            containers[0]["resources"]["limits"]["memory"] = mem
+        if cpu:
+            cpu = float(cpu)/1024
+            containers[0]["resources"]["limits"]["cpu"] = cpu
+        headers = {'Content-Type': 'application/json'}
+        conn_rc = httplib.HTTPConnection(self.target+":"+self.port)
+        conn_rc.request('POST', '/api/'+self.apiversion+'/namespaces/default/' +
+                        'replicationcontrollers', headers=headers, body=json.dumps(js_template))
+        resp = conn_rc.getresponse()
+        data = resp.read()
+        reason = resp.reason
+        status = resp.status
+        conn_rc.close()
+        if not 200 <= status <= 299:
+            errmsg = "Failed to create Replication Controller:{} {} {} - {}".format(
+                name, status, reason, data)
+            raise RuntimeError(errmsg)
+        create = False
+        for _ in xrange(30):
+            if not create and self._get_rc_status(name) == 404:
+                time.sleep(1)
+                continue
+            create = True
+            rc = self._get_rc_(name)
+            if ("observedGeneration" in rc["status"]
+                    and rc["metadata"]["generation"] == rc["status"]["observedGeneration"]):
+                break
+            time.sleep(1)
+        return json.loads(data)
+
+    def create(self, name, image, command, **kwargs):
+        self._create_rc(name, image, command, **kwargs)
```

Comment:
@kmala you need to change the creation order. First the service and then the rc because with the current order the spreading of pods in the cluster is not possible.

https://github.com/GoogleCloudPlatform/kubernetes/issues/7257
https://github.com/GoogleCloudPlatform/kubernetes/issues/11144